### PR TITLE
support no label training

### PR DESCRIPTION
**Pull request by jerrywgz**
1. support loading crowded entry in coco dataset, set load_crowd=True to enable the feature
2. support no label training, set allow_empty=True to enable the feature. Faster RCNN and Mask RCNN are verified
3. support sample empty entry randomly and set empty_rato to control the ratio of empty number to total sample number 
4. there is no influence on training speed

### CODE REVIEW

File: ppdet/data/reader.py [hunk 1]
```diff
@@ -228,13 +225,12 @@ def __init__(self,
...
-                 drop_empty=True,
                  num_classes=80,
                  collate_batch=True,
                  **kwargs):
-        super(TrainReader, self).__init__(
-            sample_transforms, batch_transforms, batch_size, shuffle, drop_last,
-            drop_empty, num_classes, collate_batch, **kwargs)
+        super(TrainReader, self).__init__(sample_transforms, batch_transforms,
+                                          batch_size, shuffle, drop_last,
+                                          num_classes, collate_batch, **kwargs)
```
Comment (qingqing01) [inline ppdet/data/reader.py:233]: config里全局搜索下，有没有地方还设置了drop_empty

File: ppdet/modeling/proposal_generator/target.py [hunk 2]
```diff
@@ -22,54 +22,83 @@
...
 
 
-def label_box(anchors, gt_boxes, positive_overlap, negative_overlap,
-              allow_low_quality):
+def label_box(anchors, gt_boxes, is_crowd, positive_overlap, negative_overlap,
+              allow_low_quality, ignore_thresh):
     iou = bbox_overlaps(gt_boxes, anchors)
-    if iou.numel() == 0:
+    N_gt = gt_boxes.shape[0]
+    N_gt_crowd = paddle.nonzero(is_crowd).shape[0]
```
Comment (qingqing01) [inline ppdet/modeling/proposal_generator/target.py]: n_gt_crowd

**Review by qingqing01 — COMMENTED** [review]
MaskRCNN(FPN) 和 FasterRCNN(FPN) 都测试过不？

File: ppdet/data/reader.py (hunk 1 above)
Comment (jerrywgz) [reply ppdet/data/reader.py:233]: done, thx

File: ppdet/modeling/proposal_generator/target.py (hunk 2 above)
Comment (jerrywgz) [reply ppdet/modeling/proposal_generator/target.py]: done

**Comment by jerrywgz** [comment]
> MaskRCNN(FPN) 和 FasterRCNN(FPN) 都测试过不？

目前还只支持Faster RCNN(FPN)

**Comment by jerrywgz** [comment]
> > MaskRCNN(FPN) 和 FasterRCNN(FPN) 都测试过不？
> 
> 目前还只支持Faster RCNN(FPN)

已支持mask rcnn, faster rcnn

**Review by qingqing01 — APPROVED** [review]

File: ppdet/modeling/proposal_generator/target.py (hunk 2 above)
Comment (lanma901106) [reply ppdet/modeling/proposal_generator/target.py]: where I should add "allow_empty=True" to enable this function.

**Pull request merged**